### ISSUE TITLE
Inheritance diagrams are excessively large in 3.2.0 docs

### ISSUE THREAD

**Issue by timhoffm** [matplotlib/matplotlib#16856]
Compare
https://matplotlib.org/3.2.0/api/artist_api.html#matplotlib.artist
vs
https://matplotlib.org/devdocs/api/artist_api.html#matplotlib.artist

We're using svg diagrams now (#16113). But that applies to both 3.2.0 and devdocs. So that alone cannot explain it. Are the versioned docs built differently?

**Comment by QuLogic** [comment]
I built 3.2.1 docs using a Fedora container, while 3.2.0 docs were built by (I think) @tacaswell using Arch? The devdocs are built on CircleCI and auto-deployed from there. Since both 3.2.z were built on up-to-date (or more so than CI) machines, it's probably some dependency change. I cannot reproduce this problem using my normal conda environment, so that does limit things to whatever's different between that and the container.

**Comment by tacaswell** [comment]
@QuLogic is correct about how the 3.2.0 docs got built.

And the links are broken in all three (devdocs, 3.2.0, 3.2.1).  The sphinx version is different on all three (devdocs in 2.4.4 and the newest).  I'm going to try rebuilding locally to see if that fixes atleast the size problem.

**Comment by tacaswell** [comment]
It it huge locally for me as well with

```
jupiter@19:03  ➤  dot -v
dot - graphviz version 2.42.3 (0)

```

Looks like there is a "transform" attribute embedded in the svg that is the source of the problem (manually changing it down in the 3.2.1 docs to match what it is in the dev docs more-or-less works).

Comparing 3.2.1 (-) to devdocs (+) (and truncating what looks like to just a bunch of numbers changing)

```diff
--- inheritance-9cf3b04035010a600653b1b41148c9a4774715d8.svg	2020-03-18 20:54:55.000000000 -0400
+++ inheritance-dbbce4a80665182ff1d05720d9f0da0eb41f1e36.svg	2020-03-20 14:06:24.000000000 -0400
@@ -1,77 +1,77 @@
 <?xml version="1.0" encoding="UTF-8" standalone="no"?>
 <!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN"
  "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
-<!-- Generated by graphviz version 2.43.0 (0)
+<!-- Generated by graphviz version 2.40.1 (20161225.0304)
  -->
-<!-- Title: inheritanceee34a5a9e5 Pages: 1 -->
-<svg width="575pt" height="864pt"
- viewBox="0.00 0.00 574.72 864.00" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
-<g id="graph0" class="graph" transform="scale(1.82 1.82) rotate(0) translate(4 1570)">
-<title>inheritanceee34a5a9e5</title>
-<polygon fill="white" stroke="transparent" points="-4,4 -4,-1570 1043,-1570 1043,4 -4,4"/>
+<!-- Title: inheritance62517ad0ab Pages: 1 -->
+<svg width="549pt" height="864pt"
+ viewBox="0.00 0.00 549.48 864.00" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
+<g id="graph0" class="graph" transform="scale(.534 .534) rotate(0) translate(4 1614)">
+<title>inheritance62517ad0ab</title>
+<polygon fill="#ffffff" stroke="transparent" points="-4,4 -4,-1614 1025,-1614 1025,4 -4,4"/>
 <!-- AitoffAxes -->
 <g id="node1" class="node">
 <title>AitoffAxes</title>
```

To pull the offending SVGs (the devdocs one might move):
```
wget matplotlib.org/3.2.1/_images/inheritance-9cf3b04035010a600653b1b41148c9a4774715d8.svg
wget matplotlib.org/devdocs/_images/inheritance-dbbce4a80665182ff1d05720d9f0da0eb41f1e36.svg
```

**Comment by QuLogic** [comment]
On Fedora 30, it's graphviz 2.40.1, but 2.42.2 in the container I used.

**Comment by QuLogic** [comment]
Seems to be a bug in graphviz https://gitlab.com/graphviz/graphviz/-/issues/1605 Best we can do is avoid those versions.

**Comment by QuLogic** [comment]
Upstream released 2.42.4 which should fix this, and is in Fedora 32, so should be available the next time I build the docs.

**Comment by timhoffm** [comment]
Should we close this then?

**Comment by QuLogic** [comment]
Well, there's actually still the question of why the links are broken (they're missing `/api/` at the beginning of the path).

**Comment by QuLogic** [comment]
Current docs now have correctly-sized diagrams. I forgot to look into the `/api/` thing though.

**Comment by tacaswell** [comment]
The size is fixed https://matplotlib.org/3.2.2/api/artist_api.html#matplotlib.artist but the url are still broken.

**Comment by QuLogic** [comment]
The linking seems like it might be related to https://github.com/sphinx-doc/sphinx/issues/3176

**Comment by QuLogic** [comment]
The `/api/` links were fixed by #17913.
